ingestion/src/metadata/utils/dispatch.py

```diff
@@ -54,3 +56,53 @@ def inner(fn):
 
     Register = namedtuple("Register", ["add", "registry"])
     return Register(add, registry)
+
+
+def valuedispatch(func) -> Callable:
```

Comment:
Could you please explain the rationale behind this? I see it being used with enums and not consistently passing the value vs. the enum object

```python
@get_system_metrics_for_dialect.register(Dialects.BigQuery)
# ...
@_get_metrics.register(MetricTypes.Table.value)
```

Could you please explain why is the extra logic needed instead of just using the `enum_register`? Thanks!

Reply:
It always passes strings. If it does not the dispatch function will throw an error. `MetricTypes ` inherit from `enum.Enum` while Dialects inherits from `sqlalchemy.sql.sqltypes.Enum`. The first one returns an enum object so we need to access its value through the `value` method while the second one returns a `str`.

Reply:
https://github.com/open-metadata/OpenMetadata/blob/240babb092f2a864adc247a8c4128b8a08b9da7e/ingestion/src/metadata/utils/dispatch.py#L73-L74

Reply:
aaaaah okok so basically we're dispatching based on strings. got it, thank you

Reply:
sorry you were my first review of the day and coffee did not kick in 😂